### PR TITLE
Add Module Position above the logo

### PR DESCRIPTION
**Pull request by coolcat-creations**
Pull Request for Issue "Add positions to Joomla 4 Cassiopeia Default template." #27193

### Summary of Changes
I added a simple topbar position because I think it will be very useful for many even if cassiopeia does not use it.
The Footer position seemed to be fixed already so you can add multiple modules there.

### Testing Instructions
Apply the patch and see if you can add modules at the topbar position. 

### Actual result BEFORE applying this Pull Request
Topbar position does not exist

### Expected result AFTER applying this Pull Request
Topbar position exist

### Documentation Changes Required
Yes, mentioning the module position maybe

### CODE REVIEW

File: templates/cassiopeia/index.php [hunk 1]
```diff
@@ -116,6 +116,14 @@
 	echo ($this->direction == 'rtl' ? ' rtl' : '');
 ?>">
 	<header class="header container-header full-width <?php echo $stickyHeader; ?>">
+
+		<?php if ($this->countModules('topbar')) : ?>
+			<div class="container-topbar">
+					<jdoc:include type="modules" name="topbar" style="none" />
```
Comment (brianteeman) [inline templates/cassiopeia/index.php]:
```suggestion
			<jdoc:include type="modules" name="topbar" style="none" />
```

code style

**Comment by ChristineWk** [comment]
Hi Elisa,
With Patch, the Topbar position exist.
For testing purposes, I created 1 custom-module, Title to show and small text. But can't see? [with Preparing content also tried)

screen shot 2020-12-19 at 15 29 43



<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by coolcat-creations** [comment]
> Hi Elisa,
> With Patch, the Topbar position exist.
> For testing purposes, I created 1 custom-module, Title to show and small text. But can't see? [with Preparing content also tried)
> 
> screen shot 2020-12-19 at 15 29 43
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/31731](https://issues.joomla.org/tracker/joomla-cms/31731).

Thank you thats strange - thanks for reporting - I will look into it.

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on 4cf8f1f46e72f376180ee4a80a47e090fc5e9007<br /><br />Done:) With more moduls type<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by drmenzelit** [comment]
The idea is good, but I think we need also some styling for the new position and a meaningful landmark for a11y.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by chmst** [comment]
I am not in favour of this change.
It is correct, that many websites use a topbar for example for language selection or important information. But this is individual and must be designed carefully, especially for small devices.

It might be a a11y issue. Modules above the navigation have to be recognized and skipped for non mouse users.   <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by ChristineWk** [comment]
With other module types it works, but I also think, it needs some styling. 

screen shot 2020-12-19 at 21 24 27<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by infograf768** [comment]
If patch is accepted, shall not we add a new lang string in the sys.ini ?

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 4cf8f1f46e72f376180ee4a80a47e090fc5e9007<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by ChristineWk** [comment]
OK, I tried again. Custom Module is possible. I didn't see the text, as the default color is: #212529; :-)
temporarily changed to #fff;

screen shot 2020-12-20 at 13 51 09

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

About the other points, I don't know.

**Comment by chmst** [comment]
After dicussion with JAT: As the new position is insinde a landmark (header) id can be used and is no a11y issue. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by gostn** [comment]
I have tested this item :white_check_mark: successfully on 4cf8f1f46e72f376180ee4a80a47e090fc5e9007<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by infograf768** [comment]
**This can't be tested successfully** until the new lang string is added: 
https://github.com/joomla/joomla-cms/pull/31731#issuecomment-748571633

**Comment by gostn** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by coolcat-creations** [comment]
thanks, I have lifetime issues getting notifications from github, going to add the string.

**Comment by richard67** [comment]
@coolcat-creations I was so free to add the language string for you and resolve a merge conflict resulting from this by updating your branch to latest 4.0-dev.

@ChristineWk @adj9 Could you test this PR again? It just needs to check that we have a proper language sting being used in the  module position selection for a module instead of "topbar".

**Comment by richard67** [comment]
@infograf768 I've added a language string. Maybe you can give it a test in case if one of the previous testers is not available anymore?

**Comment by ChristineWk** [comment]
@richard67 
My testsite is available :-)

To be sure, it shows now:

screen shot 2021-04-18 at 11 47 57

Still have the previous module and in addition now,  the second one (Search). But space in between?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by richard67** [comment]
@ChristineWk Could you start with a new installation of the package for this PR? Compared to the previous test, in this PR only the new language string `TPL_CASSIOPEIA_POSITION_TOPBAR="Top Bar"` has been added, but this PR has also been updated to latest changes in the 4.0-dev branch after that, and there might have been made changes which have an impact on CSS here, too.

I know it's a bit work, but we can be sure we don't see old, meanwhile solved CSS issues when testing.

For the new language string just check if for the module the position "Top Bar" is shown instead of "topbar" like it was in previous tests.

**Comment by ChristineWk** [comment]
Above Status: 4.0.0.beta8-dev from 11april2021. But OK, will install Prebuild Update Package now.

**Comment by ChristineWk** [comment]
@richard67 
got the same result (as on screenshot).
Status now: Site 	File 	No Problems 	4.0.0-2021-02-28 	‎4.0.0-beta8-dev+pr.31731 
Checked also: https://github.com/joomla/joomla-cms/tree/4.0-dev/administrator/components/com_admin/sql/updates/mysql<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by richard67** [comment]
@ChristineWk So I understand right and it needs some styling for the new module position? Or is it an issue also without this PR that there is too much spacing, e.g. when putting the custom module on position "Search" and order so it comes before the "search" module? If it's also an issue without this PR, I'd say it's a good test here, but if not, we should ping someone for adding some styling.

**Comment by ChristineWk** [comment]
@richard67 
Too much spacing when using two modules on this position.

screen shot 2021-04-18 at 12 47 36

<p>&nbsp;</p>

Position: Top Bar [topbar] works. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by richard67** [comment]
@ChristineWk But the markup `<p>&nbsp;</p>`comes from the custom module's content, or not? What happens if you remove that?

File: language/en-GB/tpl_cassiopeia.sys.ini [hunk 2]
```diff
@@ -19,4 +19,5 @@ TPL_CASSIOPEIA_POSITION_SIDEBAR-LEFT="Sidebar-left"
 TPL_CASSIOPEIA_POSITION_SIDEBAR-RIGHT="Sidebar-right"
 TPL_CASSIOPEIA_POSITION_TOP-A="Top-a"
 TPL_CASSIOPEIA_POSITION_TOP-B="Top-b"
+TPL_CASSIOPEIA_POSITION_TOPBAR="Top Bar"
```
Comment (Quy) [inline language/en-GB/tpl_cassiopeia.sys.ini:22]: Should it be `Top-bar` or `Topbar`?

**Comment by ChristineWk** [comment]
@richard67 
Yes, Is saw above markup .... (that's why I made a screenshot :.)
Will check the custom module's content x adv asap.
(was a previous generated custom content) thks for waiting :-) <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

File: templates/cassiopeia/index.php [hunk 3]
```diff
@@ -115,6 +115,14 @@
...
 ?>">
 	<header class="header container-header full-width <?php echo $stickyHeader; ?>">
+
+		<?php if ($this->countModules('topbar')) : ?>
+			<div class="container-topbar">
+			<jdoc:include type="modules" name="topbar" style="none" />
+			</div>
+		<?php endif; ?>
+
+
```
Comment (Quy) [inline templates/cassiopeia/index.php]:
```suggestion
```

File: language/en-GB/tpl_cassiopeia.sys.ini (hunk 2 above)
Comment (richard67) [reply language/en-GB/tpl_cassiopeia.sys.ini:22]: "Top Bar" was what Google translator suggested to me as correction for the English when I had tried "Topbar". With "Sidebar" Google translator was ok when I cross-checked.

Maybe @brianteeman can have a look?

Comment (richard67) [reply language/en-GB/tpl_cassiopeia.sys.ini:22]: @Quy Or did you mean the name and not the value of the language string?

Comment (Quy) [reply language/en-GB/tpl_cassiopeia.sys.ini:22]: The value. The suggested change is to be consistent with the the format of the other strings.

Comment (Quy) [reply language/en-GB/tpl_cassiopeia.sys.ini:22]: The other strings don’t contain a space.

Comment (richard67) [reply language/en-GB/tpl_cassiopeia.sys.ini:22]: What is better English? "Topbar"? Or "Top-bar"? Feel free to suggest and commit a change. Elisa will be ok with it, I think.

**Comment by ChristineWk** [comment]
@richard67 

a) OK, I removed the markup `<p>&nbsp;</p>`  in HTML Editor. It was automatically done :-)
b) Will now remove Prebuild Package and should load new one?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by richard67** [comment]
> b) Will now remove Prebuild Package and should load new one?
 
 @ChristineWk Why that?

**Comment by ChristineWk** [comment]
@richard67 

Due of changes from @Quy? Just reading and reading :-)

**Comment by richard67** [comment]
@ChristineWk No, that was just a code style change. There might come a small change in the language string, also not really a big thing which requires a new test. Main question now is: Is the spacing issue still there after the change in the markup?

**Comment by ChristineWk** [comment]
> 
> 
> @ChristineWk No, that was just a code style change. There might come a small change in the language string, also not really a big thing which requires a new test. Main question now is: Is the spacing issue still there after the change in the markup?

Aha, I thought, that there is new test required. 
No :-) As I wrote, it was seen under HTML Editor. Thks.

**Comment by richard67** [comment]
@ChristineWk Well if this PR works for you then just mark your test result again in the issue tracker. Thanks in advance, and thanks for testing.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 492b480fda744b6d32734abe7498b7b9e9f4e49d<br /><br />Now, I will install the Nightly Beta8-dev from today :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 492b480fda744b6d32734abe7498b7b9e9f4e49d<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31731">issues.joomla.org/tracker/joomla-cms/31731</a>.</sub>

**Pull request merged**